lib/synapse/service_watcher/base.rb

```diff
@@ -99,6 +102,17 @@ def validate_discovery_opts
       log.warn "synapse: warning: a stub watcher with no default servers is pretty useless" if @default_servers.empty?
     end
 
+    def filter_backends_by_label(backends, label_filter)
+      filtered_backends = []
+      backends.each do |backend|
+        backend_labels = backend['labels'] || {}
+        if label_filter['condition'] == 'equals' and backend_labels[label_filter['label']] == label_filter['value']
```

Comment:
oops, i should have caught this earlier. using `and` is not a good idea in ruby, you should always use `&&` because `and` has odd precedence rules. i'm working on this section of code, so i'll just fix it in my PR.